apps/labs/posts/automated-accessibility-testing-with-bokeh.md

```diff
@@ -0,0 +1,178 @@
+could start making fixes to the pertinent parts of Bokeh. For every fix, there
+would need to be an accompanying test to avoid regression later on.
+The problem, though, was that much of Bokeh runs on custom-built solutions. It
+had its own setup for almost everything. It required specific types of
+environments to match its use cases, it had a custom compilation, build & run
+tooling, and they weren't as thoroughly documented. Bokeh itself has good
+documentation for users, but the internal documentation of how things worked
+wasn't really there. What this meant for my project was that it would be more of
+research work, trying to come up with something to fit into what Bokeh had, with
+the least amount of friction. This, while not sacrificing functionality.
+I'll have to mention that prior to this project, there was no provision for
+Accessibility testing in the Bokeh codebase, and the closest we had to that was
+removed sometime in the past because it had been unused and unmaintained for so
+long. The primary goal of my work would thus be to move Bokeh from 0 to 1 and
+give feedback and findings that could be built upon. Essentially, the spark that
+starts the fire.
+
+## How Bokeh works under the hood
+
+Talking about the scope of what I worked on and the accompanying complexities
+wouldn't make as much sense if I didn't explain the mechanism behind Bokeh. To
+the typical user who needs to get work done, you just install the package, type
+some commands as instructed by the docs, and then run. In doing this, if you
+look closely at your file directly you'll realize that a matching HTML file has
+been created and automatically opens in your browser. Voila! You get an
+interactive plot. With a few lines of Python code, you've done some apparent
+magic.
+This is where things start to get fun. At the end of the day, Javascript is what
+gives interactivity to any HTML page, so the fact that you can interact with the
+beautiful data visualization on an HTML page implies the existence of CSS and
+Javascript. This means Python code is responsible for generating HTML, CSS, and
+Javascript. For this to happen with any tool, there needs to be a shared data
+model between what would be the backend (Python) and what would be client-side
+(what you see on the browser). Bokeh has a brilliant way of doing this, and I'll
+attach a picture from their [SciPy 2023
+tutorial](https://www.youtube.com/watch?v=G0Yc3ck4lC8) to illustrate:
+
+<img alt="Image of Bokeh data model flow represented as a flow chart moving from
+Python to JSON, to Javascript, and then to final HTML output"
+src="/posts/automated-accessibility-testing-with-bokeh/bokeh_bokehjs.jpg"
+style={{padding: '10px 20px'}}/>
+
+The implication of this is that virtually all things on the Python side should
+be representable on a Javascript (for the most part).
+
+## Testing tools
+
+For the right blend of power, ease of use, and modernity, my mentor
```

Comment:
There's no transition here. As a reader, this feels like a sharp turn. One second we were talking about shared models, now we're talking about testing? In the reader's mind: "what's the connection? I'm lost"